src/components/side_panel/chart/chart_with_axis/design_panel.ts

```diff
@@ -131,4 +142,64 @@ export class ChartWithAxisDesignPanel extends Component<Props, SpreadsheetChildE
     const dataSets = this.props.definition.dataSets;
     return dataSets[this.state.index]?.label || this.getDataSeries()[this.state.index];
   }
+
+  toggleDataTrend(display: boolean) {
+    const dataSets = this.props.definition.dataSets;
+    if (!dataSets?.[this.state.index]) {
+      return;
+    }
+    dataSets[this.state.index] = {
+      ...dataSets[this.state.index],
+      trend: {
+        type: "polynomial",
+        order: 3,
```

Comment:
There is no proof that the 3rd order polynomial would be a better choice but it will fit most dataset than a linear model. Anyway, going with the simplest model as default seem legit.

For the automatic choice, it would be realy heavy as we would have to make some matrix computation for each model to be compared for a data serie and then choose the best. I have another idea that will require some IA trained to automatically detect some parameters on the dataset and then apply the correct model (which would then be an option "Best fit"). It would be something not heavy at all (once trained, of course), but it will be for another task, one day :)

Reply:
Let's go for linear :)